### PR TITLE
Alerting Docs: Monitor your high availability setup

### LINKED ISSUE #85913: Alering: Document how to verify that Alertmanger HA gossip is working
Docs here:
https://grafana.com/docs/grafana/latest/alerting/set-up/configure-high-availability/

We should add a few lines about how to verify that HA is working, something along the lines of:

`alertmanager_cluster_members` should be the number of replicas
`alertmanager_peer_position` should be different depending on the replica: for two replicas, one should have a `alertmanager_peer_position==0` and the other `alertmanager_peer_position==1`

### PR DESCRIPTION
**Pull request by ppcano**
Document how to verify that Alertmanger HA gossip is working. Closes https://github.com/grafana/grafana/issues/85913 

Also, refer to [new alerting high availability examples using Docker Compose](https://github.com/grafana/alerting-ha-docker-examples/) - it would be ideal to also review these examples.

### CODE REVIEW

File: docs/sources/alerting/set-up/configure-high-availability/_index.md [hunk 1]
```diff
@@ -56,6 +67,8 @@ Since gossiping of notifications and silences uses both TCP and UDP port `9094`,
    By default, it is set to listen to all interfaces (`0.0.0.0`).
 1. Set `[ha_peer_timeout]` in the `[unified_alerting]` section of the custom.ini to specify the time to wait for an instance to send a notification via the Alertmanager. The default value is 15s, but it may increase if Grafana servers are located in different geographic regions or if the network latency between them is high.
 
+For a demo, see this [example of alerting high availability using Memberlist and Docker Compose](https://github.com/grafana/alerting-ha-docker-examples/tree/main/memberlist).
```
Comment (brendamuir) [inline docs/sources/alerting/set-up/configure-high-availability/_index.md:70]:
```suggestion
For a demo, see this [example](https://github.com/grafana/alerting-ha-docker-examples/tree/main/memberlist) of alerting high availability using Memberlist and Docker Compose.
```

File: docs/sources/alerting/set-up/configure-high-availability/_index.md [hunk 2]
```diff
@@ -149,3 +150,40 @@ The following metrics can be used for meta monitoring, exposed by the `/metrics`
    ha_peer_timeout = 15s
    ha_reconnect_timeout = 2m
    ```
+
+## Monitor your high availability setup
+
+When running multiple Grafana instances, all alert rules are evaluated on every instance. This multiple evaluation of alert rules is visible in the [state history](ref:state-history) and provides a straightforward way to verify that your high availability configuration is working correctly.
+
+You can also confirm your high availability setup by monitoring alertmanager metrics exposed by Grafana:
```
Comment (brendamuir) [inline docs/sources/alerting/set-up/configure-high-availability/_index.md:158]:
```suggestion
You can also confirm your high availability setup by monitoring Alertmanager metrics exposed by Grafana.
```

File: docs/sources/alerting/set-up/configure-high-availability/_index.md [hunk 3]
```diff
@@ -149,3 +150,40 @@ The following metrics can be used for meta monitoring, exposed by the `/metrics`
...
+| alertmanager_cluster_messages_publish_failures_total | Total number of messages that failed to be published.                                                          |
+| alertmanager_cluster_pings_seconds                   | Histogram of latencies for ping messages.                                                                      |
+| alertmanager_cluster_pings_failures_total            | Total number of failed pings.                                                                                  |
+| alertmanager_peer_position                           | Position the Alertmanager instance believes it's in. The position determines a peer's behavior in the cluster. |
+
+You can confirm the number of Grafana instances in your alerting high availability setup by querying the `alertmanager_cluster_members` and `alertmanager_peer_position` metrics.
+
+But note that these alerting high availability metrics are exposed via the `/metrics` endpoint in Grafana, and are not automatically collected or displayed. If you have a Prometheus instance connected to Grafana, you can add a `scrape_config` to scrape Grafana metrics and then query these metrics in Explore.
```
Comment (brendamuir) [inline docs/sources/alerting/set-up/configure-high-availability/_index.md:174]:
```suggestion
Note that these alerting high availability metrics are exposed via the `/metrics` endpoint in Grafana, and are not automatically collected or displayed. If you have a Prometheus instance connected to Grafana, add a `scrape_config` to scrape Grafana metrics and then query these metrics in Explore.
```

File: docs/sources/alerting/set-up/configure-high-availability/_index.md [hunk 4]
```diff
@@ -149,3 +150,40 @@ The following metrics can be used for meta monitoring, exposed by the `/metrics`
...
+  scrape_timeout: 10s
+  metrics_path: /metrics
+  scheme: http
+  follow_redirects: true
+  static_configs:
+    - targets:
+        - grafana:3000
+```
+
+For more details about monitoring alerting metrics, refer to [Alerting meta-monitoring](ref:meta-monitoring). For a demo, see the [alerting high availability examples using Docker Compose](https://github.com/grafana/alerting-ha-docker-examples/).
```
Comment (brendamuir) [inline docs/sources/alerting/set-up/configure-high-availability/_index.md:189]:
```suggestion
For more information on monitoring alerting metrics, refer to [Alerting meta-monitoring](ref:meta-monitoring). For a demo, see [alerting high availability examples using Docker Compose](https://github.com/grafana/alerting-ha-docker-examples/).
```

**Review by brendamuir — APPROVED** [review]

**Comment by ppcano** [comment]
@brendamuir, did another round of minor changes after considering https://github.com/grafana/grafana/issues/88446

**Review by brendamuir — APPROVED** [review]

File: docs/sources/alerting/set-up/configure-high-availability/_index.md [hunk 5]
```diff
@@ -149,3 +145,48 @@ The following metrics can be used for meta monitoring, exposed by the `/metrics`
...
+| alertmanager_cluster_messages_sent_size_total        | Total number of cluster messages received.                                                                     |
+| alertmanager_cluster_messages_publish_failures_total | Total number of messages that failed to be published.                                                          |
+| alertmanager_cluster_pings_seconds                   | Histogram of latencies for ping messages.                                                                      |
+| alertmanager_cluster_pings_failures_total            | Total number of failed pings.                                                                                  |
+| alertmanager_peer_position                           | Position the Alertmanager instance believes it's in. The position determines a peer's behavior in the cluster. |
+
+You can confirm the number of Grafana instances in your alerting high availability setup by querying the `alertmanager_cluster_members` and `alertmanager_peer_position` metrics.
```
Comment (stevesg) [inline docs/sources/alerting/set-up/configure-high-availability/_index.md:175]: A comment here about what to expect from the `alertmanager_peer_position` metric I think would be helpful, something like:

> Under normal operation, the `alertmanager_peer_position` should be different for Grafana instance; the peers should be incrementally numbered starting at zero, for example, with two instances, one instance should have peer position `0` and the other should have peer position `1`.

**Review by stevesg — APPROVED** [review]
LGTM, I suggest one extra comment but I'll let you decide if it's too much detail or not.

File: docs/sources/alerting/set-up/configure-high-availability/_index.md (hunk 5 above)
Comment (ppcano) [reply docs/sources/alerting/set-up/configure-high-availability/_index.md:175]: Yes! Addressed by https://github.com/grafana/grafana/pull/92063#pullrequestreview-2260149953

**Pull request merged**